config/database.yml

```diff
@@ -1,25 +1,22 @@
-development:
+default: &default
   adapter: postgresql
-  database: local_links_manager_development
+  encoding: unicode
   template: template0
-  encoding: utf8
-  variables:
-    work_mem: 10MB
-    maintenance_work_mem: 128MB
+
+development:
+  <<: *default
+  database: local-links-manager_development
 
 # Warning: The database defined as "test" will be erased and
 # re-generated from your development database when you run "rake".
 # Do not set this db to the same as development or production.
 test: &test
-  adapter: postgresql
-  database: local_links_manager_test<%= ENV['TEST_ENV_NUMBER'] %>
-  template: template0
-  encoding: utf8
-  variables:
-    work_mem: 1MB
+  <<: *default
+  database: local-links-manager_test<%= ENV['TEST_ENV_NUMBER'] %>
 
 cucumber:
   <<: *test
 
 production:
+  <<: *default
   url: <%= ENV['DATABASE_URL'] %>
```

Comment:
I haven't seen this approach before.  How does this work with the env variables hash set in puppet?  I notice that content-tagger uses username/password/database.